File: hw/segarra/u3/HW28SingletonAndAbstractFactory/HW28SingletonJavaScript/ec.edu.espe.singleton/controller/CalculatorController.js
class CalculatorController {
    constructor(model, view) {
        this.model = model;
        this.view = view;
    }

    calculateTotalWithTax(amount) {
        const tax = this.model.calculateTax(amount);
        const total = amount + tax;
        this.view.displayTotal(total);
    }
}

File: hw/Vaca/u3/HW28SingletonAndAbstractFactory/HW29SingletonJava/src/main/java/ec/edu/espe/singleton/controller/CalculatorController.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.singleton.controller;

import ec.edu.espe.singleton.model.USTax;
/**
 *
 * @author MSI-PULSE
 */


public class CalculatorController {
    public double getTotalSale(double saleAmount) {
        USTax tax = USTax.getInstance();
        return saleAmount + tax.calculateTax(saleAmount);
    }
}